src/WebJobs.Extensions.DurableTask/ProtobufUtils.cs

```diff
@@ -617,5 +618,41 @@ internal static P.PurgeInstancesResponse CreatePurgeInstancesResponse(PurgeResul
                     throw new NotSupportedException($"Deserialization of {operationResult.ResultTypeCase} is not supported.");
             }
         }
+
+        /// <summary>
+        /// Converts a <see cref="Dictionary{TKey, TValue}"/> of <c>&lt;string, object?&gt;</c> to a <see cref="MapField{String, Value}"/> for gRPC transmission.
+        /// </summary>
+        /// <param name="input">The dictionary to convert.</param>
+        /// <returns>The corresponding <see cref="MapField{String, Value}"/>, or <c>null</c> if input is <c>null</c>.</returns>
+        internal static MapField<string, Value> ConvertDictionaryToStructMap(Dictionary<string, object?> input)
```

Comment:
nit: consider renaming this method to match the return type.

```suggestion
        internal static MapField<string, Value> ConvertDictionaryToMapField(Dictionary<string, object?> input)
```